docs/framework-overview.md

```diff
@@ -0,0 +1,398 @@
+
+<figure markdown>
+  ![EasyBuild Framework package structure](img/framework-package-diagram-4.7.1.svg)
+  <figcaption>Package structure of EasyBuild Framework v4.7.1: easybuild.framework in blue and easybuild.tools in orange.</figcaption>
+</figure>
+
+## Workflow Overview {: #workflow_overview }
+
+
+The following diagram shows the main steps carried out by EasyBuild for a
+regular software installation with the `eb` command:
+
+```mermaid
+stateDiagram-v2
+    easybuild.main --> tools
+    robot --> build_install
+    build_install --> easyblock
+    
+    state "easybuild.tools" as tools{
+        [*] --> options
+        options --> parse_ec
+        process_ec --> robot
+        
+        state "easybuild.tools.options
+        • parsing of options
+        • set_up_configuration( )" as options
+        
+        state "easybuild.tools.robot
+        • resolve_dependencies( )" as robot
+    }
+
+    state "build_and_install_software()" as build_install
+    
+    state "easybuild.framework.easyblock" as easyblock {
+        [*] --> build_install_one
+        build_install_one --> get_easyblock
+        get_easyblock --> easyblock_class
+
+        state "easybuild.framework.easyblock
+        • build_and_install_one()" as build_install_one
+        
+        state "class EasyBlock
+        • run_all_steps()" as easyblock_class
+    }
+
+    state "easybuild.framework.easyconfig" as easyconfig{
+        parse_ec --> process_ec
+
+        state "easybuild.framework.easyconfig.tools
+        • parse_easyconfigs()" as parse_ec
+
+        state "easybuild.framework.easyconfig.easyconfig
+        • process_easyconfig()" as process_ec
+
+        state "easybuild.framework.easyconfig.easyconfig
+        • get_easyblock_class()" as get_easyblock
+    }
+    
+```
+
+The process starts with `easybuild.tools` taking care of configuration options
+passed to EasyBuild. Either through the command line, configuration files or in
+the environment.
+
+The scope of the EasyBuild execution is ultimately defined by the target
+easyconfig files given in the `eb` command. They will define the number of
+actions to carry out, such as package installations, fetching of sources or
+opening of PRs. Therefore, target easyconfig files are parsed early in the
+process, just after the runtime options have been resolved. The tooling to
+parse `EasyConfig` files is located in a specific `tools` module in
+`easybuild.framework.easyconfig.tools`, while the `EasyConfig` class is found
+in `easybuild.framework.easyconfig.easyconfig`.
+
+Once the easyconfig files are properly parsed, a common action is to resolve
+the package dependencies for those installations with the
+[`--robot`][use_robot] option. This is handled in `easybuild.tools.robot` which
+provides methods to search easyconfig files, resolve dependencies and check
+conflicts between packages.
+
+The *robot* step returns an ordered list of all software packages that have to
```

Comment:
I would say "step" here, to keep that term for the steps performed by `EasyBlock`:

```suggestion
The `resolve_dependencies` function returns an ordered list of all (parsed) easyconfigs that have to